src/Adapter/Shipment/QueryHandler/GetOrderShipmentsForViewingHandler.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Shipment\QueryHandler;
+
+use PrestaShop\Decimal\DecimalNumber;
+use PrestaShop\PrestaShop\Core\CommandBus\Attributes\AsQueryHandler;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\Exception\ShipmentNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\Query\GetOrderShipments;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\QueryHandler\GetOrderShipmentForViewingHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\QueryResult\OrderShipment;
+use PrestaShopBundle\Entity\Repository\ShipmentRepository;
+use Throwable;
+
+#[AsQueryHandler]
+class GetOrderShipmentsForViewingHandler implements GetOrderShipmentForViewingHandlerInterface
+{
+    public function __construct(
+        private readonly ShipmentRepository $repository,
+    ) {
+    }
+
+    /**
+     * @param GetOrderShipments $query
+     *
+     * @return OrderShipment[]
+     */
+    public function handle(GetOrderShipments $query)
+    {
+        $shipments = [];
+        $orderId = $query->getOrderId()->getValue();
+
+        try {
+            $result = $this->repository->findByOrderId($orderId);
+        } catch (Throwable $e) {
+            throw new ShipmentNotFoundException(sprintf('Could not find shipment for order with id "%s"', $orderId), 0, $e);
+        }
+
+        foreach ($result as $shipment) {
+            $shipments[] = new OrderShipment(
+                $shipment->getId(),
+                $shipment->getOrderId(),
+                $shipment->getCarrierId(),
+                $shipment->getAddressId(),
+                new DecimalNumber((string) $shipment->getShippingCostTaxExcluded()),
+                new DecimalNumber((string) $shipment->getShippingCostTaxIncluded()),
+                count($shipment->getProducts()),
```

Comment:
```suggestion
                $shipment->getProducts()->count(),
```
It may look like nothing but I believe it's a good practice, because with lazy collection Doctrine may be able to optimize and get the count without requesting all the content

Reply:
Does it need to be there? If someone wants to get the count, he can count it in the template or code.